six/modules/c++/six.sidd/include/six/sidd/SFA.h

```diff
@@ -205,8 +201,10 @@ struct SFAPolyhedralSurface : public SFASurface
         SFASurface(TYPE_NAME)
     {
     }
-    virtual ~SFAPolyhedralSurface()
+
+    virtual SFAPolyhedralSurface* clone() const
     {
+        return new SFAPolyhedralSurface(*this);
     }
 
     std::vector<mem::ScopedCopyablePtr<SFAPolygon> > patches;
```

Comment:
Made the edits. I’m pretty sure the ScopedCopyablePtr<SFALineString> on line 285 can stay as is, because there are no ‘extra’ fields in the two subclasses to be sliced off, just a different TYPE_NAME value.

From: Adam Sylvester [mailto:notifications@github.com]
Sent: Friday, February 12, 2016 2:29 PM
To: ngageoint/six-library <six-library@noreply.github.com>
Cc: Jonathan Means <Jonathan.Means@mdaus.com>
Subject: Re: [six-library] Fix object slicing issue with Annotations, clean up some warnings (#48)


In six/modules/c++/six.sidd/include/six/sidd/SFA.h<http://cp.mcafee.com/d/1jWVIp4zqb3XXdTSm3hOrKrpop73zhO--rv7cIcECzAQsCQrLnusd7dPhOYrhhphjvd7bz3VIhH3pAZxOLX2walgRKDbCRkdrFOVJxVdDO3z_nV5BBZB5_HTbFEzzhPOrP2b3zTbnhIyyHtNDBgY-F6lK1FJ4SCrLOb2rPUV5xcQsCXCOsVHkiP2cFASO7bV0zF7WOxfai3OVYy4tymNpIY_67wcgBiPtaiH0ly-CeussuudMH7RzOMa8OCjr8sKroh79CMnWhEwdbt91Vs-h2eNb6BQQg2YwYQgl82fSCq8a7OJa14QgiFhLPh0oBcCq8124aPSsGME0pEwIv-Md43qNEwFlnPh0oSc9g8Qd41EE0NxUQkCTzhOr8Ie8>:

>      {

> +        return new SFAPolyhedralSurface(*this);

>      }

>

>      std::vector<mem::ScopedCopyablePtr<SFAPolygon> > patches;

Actually looks like there are several other spots... you'll want to look at each one to see if it's going to slice or not.

—
Reply to this email directly or view it on GitHub<http://cp.mcafee.com/d/5fHCNEi4zqb3XXdTSm3hOrKrpop73zhO--rv7cIcECzAQsCQrLnusd7dPhOYrhhphjvd7bz3VIhH3pAZxOLX2walgRKDbCRkdrFOVJxVdDO3z_nV5BBZB5_HTbFEzzhPOrP2b3zTbnhIyyHtNDBgY-F6lK1FJcSCrLOb2rPUV5xcQsCXCOsVHkiP2cFASO7bV0zF7WOxfai3OVYy4tymNpIY_67ItAvZ0cy-CeussuudMH7RzOMa8OCjr8sKroh79CMnWhEwdbt91Vs-h2eNb6BQQg2YwYQgl82fSCq8a7OJa14QgiFhLPh0oBcCq8124aPSsGME0pEwIv-Md43qNEwFlnPh0oSc9g8Qd41EE0NxUQkCTzhOrfn0Fhl4H>.


The information contained in this communication is confidential, is intended only for the use of the recipient(s) named above, and may be legally privileged. If the reader of this message is not the intended recipient, you are hereby notified that any dissemination, distribution, or copying of this communication is strictly prohibited. If you have received this communication in error, please re-send this communication to the sender and delete the original message or any copy of it from your computer system.